File: src/problem042/solution042.py
import math


def digitsum(str_):
    return sum([ord(chr_) - 64 for chr_ in str_])


def is_triangle_num(num):
    idx = (-1 + math.sqrt(1 + 8 * num)) / 2
    return idx >= 0 and int(idx) == idx


def main():
    words = open('words.txt').read().split(',')
    wordvals = [digitsum(word.strip('"')) for word in words]
    print(len([val for val in wordvals if is_triangle_num(val)]))

if __name__ == '__main__':
    main()

File: src/problem042/solution042.hs
import Data.Char (ord)
import Data.Text (pack, unpack, replace, splitOn)

main = do
    raw <- readFile "words.txt"
    print . length . filter isTriangle . map digitsum . map unpack . map (replace (pack "\"") (pack "")) . splitOn (pack ",") $ pack raw
digitsum = sum . map (\x -> (ord x) - 64)
isTriangle n = ceiling idx == floor idx where idx = (sqrt (1 + 8 * (fromIntegral n)) - 1) / 2